### ISSUE TITLE
Unable to instantiate in asp.net core 5.0 project

### ISSUE THREAD

**Issue by jessicah** [toddams/RazorLight#383]
**Describe the bug**
`services.AddRazorLight()` doesn't appear sufficient to register types needed for DI to work.

**To Reproduce**
```services.AddSingleton<RazorLightOptions>(new RazorLightOptions());
services.AddRazorLight(() =>
{
	return new RazorLightEngineBuilder()
		.UseFileSystemProject("~/repo")
		.UseMemoryCachingProvider()
		.Build();
});
```

**Expected behavior**
RazorLight dependencies registered and able to run project.

**Information (please complete the following information):**
 - OS: Windows 10
 - Platform: .NET Core 5.0 (ASP.NET Core)
 - RazorLight: 2.0.0-beta9
 - Are you using the OFFICIAL RazorLight package? Yes
 - Visual Studio: Visual Studio Community 16.8.0 Preview 5.0

**Additional context**
System.AggregateException: 'Some services are not able to be constructed (Error while validating the service descriptor 'ServiceType: RazorLight.IEngineHandler Lifetime: Singleton ImplementationType: RazorLight.EngineHandler': Unable to resolve service for type 'RazorLight.Compilation.IRazorTemplateCompiler' while attempting to activate 'RazorLight.EngineHandler'.)'

**Comment by jessicah** [comment]
I managed to workaround the issue by hoisting the `RazorEngine` instance out of `AddRazorLight()`, then adding the following:
```
services.AddSingleton(new RazorLightOptions());
services.AddSingleton(razorEngine.Handler.Compiler);
services.AddSingleton(razorEngine.Handler.FactoryProvider);
services.AddSingleton(razorEngine.Handler.Cache);
```

But it would be nice to not have to do this... although now I'm blocked on https://github.com/toddams/RazorLight/issues/365, so I don't know if it's fully resolved yet.

**Comment by jzabroski** [comment]
@jessicah  I merged #365 fix this morning and will release this week.  Thank you for your patience.

**Comment by jzabroski** [comment]
@jessicah Please update to 2.0.0-rc.1 package.  Let me know if you run into any issues.  Closing for now.
